exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/trace_exporter/__init__.py

```diff
@@ -236,29 +236,42 @@ def _translate_data(
         sdk_resource_instrumentation_library_spans = {}
 
         for sdk_span in data:
-
-            if sdk_span.resource not in (
-                sdk_resource_instrumentation_library_spans.keys()
-            ):
+            instrumentation_library_spans_map = (
+                sdk_resource_instrumentation_library_spans.get(
+                    sdk_span.resource, {}
+                )
+            )
+            # If we haven't seen the Resource yet, add it to the map
+            if not instrumentation_library_spans_map:
+                sdk_resource_instrumentation_library_spans[
+                    sdk_span.resource
+                ] = instrumentation_library_spans_map
+            instrumentation_library_spans = (
+                instrumentation_library_spans_map.get(
+                    sdk_span.instrumentation_info
```

Comment:
Is the key here the InstrumentationInfo object itself? And could that cause problem if two objects with the different InstrumentationInfo objects exist even though they are using the same values for the name/version information?

Reply:
`InstrumentationInfo` has `__equal__` and `__hash__` for unique identification.